extensions/storages/source/deployment.ts

```diff
@@ -16,50 +19,53 @@ export function deployment (instances: Instance[], annotation: Annotation): Depe
 }
 
 function validate (instances: Instance[], annotation: Annotation): void {
-  if (annotation === undefined)
-    throw new Error('Storages annotation is required by: ' +
-      `'${instances.map((i) => i.component.locator.id).join("', '")}'`)
-
-  for (const instance of instances)
-    contains(instance, annotation)
+  assert.ok(annotation !== undefined,
+    `Storages annotation is required by: '${instances
+      .map((i) => i.component.locator.id)
+      .join("', '")}'`)
 
-  for (const ref of Object.values(annotation)) {
-    const url = new URL(ref)
+  for (const instance of instances) contains(instance, annotation)
 
-    if (!(url.protocol in providers))
-      throw new Error(`Unknown storage provider '${url.protocol}'`)
-  }
+  for (const { provider } of Object.values(annotation))
+    validateProviderId(provider)
 }
 
 function contains (instance: Instance, annotation: Annotation): void {
   for (const name of instance.manifest)
-    if (!(name in annotation))
-      throw new Error(`Missing '${name}' storage annotation ` +
+    assert.ok(name in annotation,
+      `Missing '${name}' storage annotation ` +
         `declared in '${instance.component.locator.id}'`)
 }
 
-function getSecrets (annotation: Annotation): Variable[] {
+export function validateProviderId (id: string | undefined): asserts id is keyof typeof providers {
+  assert.ok(typeof id === 'string' && id in providers, `Unknown storage provider '${id}'`)
+}
+
+function getSecrets (annotation: Readonly<Annotation>): Variable[] {
   const secrets: Variable[] = []
 
-  for (const [storage, ref] of Object.entries(annotation)) {
-    const url = new URL(ref)
-    const Provider = providers[url.protocol]
+  for (const [storage, props] of Object.entries(annotation)) {
+    const providerId = props.provider
+
+    validateProviderId(providerId)
+
+    const Provider = providers[providerId]
 
-    if (Provider.SECRETS === undefined)
-      continue
+    if (Provider.SECRETS.length === 0) continue
 
     for (const secret of Provider.SECRETS)
       secrets.push({
-        name: `TOA_STORAGES_${storage.toUpperCase()}_${secret.toUpperCase()}`,
+        name: `${SERIALIZATION_PREFIX}_${storage.toUpperCase()}_${secret.name.toUpperCase()}`,
         secret: {
           name: `toa-storages-${storage}`,
-          key: secret
+          key: secret.name,
+          optional: secret.optional
```

Comment:
Corresponding changes from `dev` should be merged to test this.
[This scenario](https://github.com/toa-io/toa/blob/f4faff0fb376e867fd675d192f42a70e320c7332/features/extensions/storages.feature#L78) should pass.

Reply:
They are merged - this PR initially was on top of that changes

Reply:
<img width="971" alt="Screenshot 2024-02-09 at 10 01 36" src="https://github.com/toa-io/toa/assets/5350898/fd2c7c5e-3805-4aa2-a142-55295012f8b4">

Reply:
[They](https://github.com/toa-io/toa/blob/148a3083a29cdcfd7452cc333ace42a79322c018/features/extensions/storages.feature#L118) are [not](https://github.com/toa-io/toa/blob/f4faff0fb376e867fd675d192f42a70e320c7332/features/extensions/storages.feature#L106).

Reply:
<img width="976" alt="Screenshot 2024-02-09 at 10 12 46" src="https://github.com/toa-io/toa/assets/5350898/3930363f-a4da-4519-af8e-1a0185aa73f8">

Rebased. Test still passing.

Reply:
I looked at the wrong scenario, sorry.